oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/web/reactive/function/client/ServletOAuth2AuthorizedClientExchangeFilterFunction.java

```diff
@@ -484,6 +494,9 @@ private ClientRequest bearer(ClientRequest request, OAuth2AuthorizedClient autho
 			response = requestAttributes.getResponse();
 		}
 		Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+		if (authentication == null && request == null && response == null) {
+			return delegate;
```

Comment:
This is misleading since it doesn't create and return a `RequestContextSubscriber` instance, but instead returns the "current" `Subscriber`. I think we should return `null` for these cases and ensure `Hooks.onLastOperator` is NOT called in `afterPropertiesSet()`. Please add a test for this as well.

Reply:
Hooks.onLastOperator() - applies global lift function on `subscribe()/block()` and it should be better installed once per application life circle.  
As Lift function is invoked only at the moment of `subscribe()` lazily (not at the moment of application context / Filter initialization) it is not possible to unsure is there web/security context or not.  

Not sure we are on the same page with case I try to fix  
One of the cases:  
I want to initialize application with some data. I have my reactive service API to do it. 
I make `ApplicationRunner` or just Application listener which invokes my reactive API on startup and call subscribe()/block(). 
So If my API implementation uses reactor context (e.g. reactive transactions, or just some custom implementation) with initialized hook from filter and without current fix I will get NPE, because `RequestContextSubscriber` will populate reactor context with `null` values (as there is no Web and Security Context) and this is forbidden.  

Another scenario - my reactive API is invoked form RabbitMQ listener and there is no web context - as the result NPE.  

So what PR fixes:  
1. it ensures that no `null` values are propagated to the reactor context.  
2. As reactor context recommended to be as small as possible (see javadocs) this fix reduces number of used entries from 4 to 1/0.  
3. `RequestContextSubscriber` replaced original context so data from other hooks could be lost. This fix ensures that `RequestContextSubscriber` will not replace all parent context but adds values to it.  
4. In described case `RequestContextDataHolder` will be created with `null` values and will be useless as it is not required at all for the operation (application initialization), so it is make sense not to populate reactor context with useless data because of p.2 and not to create additional `Subscriber` to reduce allocation and invocation stack size on each emit or `currentContext()`

Lift function can't return `null` as later we will get NPE so `delegate` subscriber looks fine to me as it means NOOP lift function.   

@jgrandja, do you still want to return `null` from lift function and create `RequestContextSubscriber` in any case with empty `RequestContextDataHolder`?